sksports/model/heart_rate.py

```diff
@@ -0,0 +1,164 @@
+"""Module to estimate heart rate from data."""
+
+# Authors: Aart Goossens
+#          Guillaume Lemaitre
+# License: MIT
+
+from __future__ import division
+
+import numpy as np
+from scipy.optimize import leastsq, minimize
+
+from sklearn.base import BaseEstimator, RegressorMixin
+from sklearn.utils import check_X_y, check_array
+from sklearn.utils.validation import check_is_fitted
+
+
+def _model(power, hr_rest, hr_max, hr_slope, hr_drift, rate_growth,
+           rate_decay):
+    """Model to predict heart-rate from power data."""
+    power_corrected = power + power.cumsum() * hr_drift
+    hr_lin = power_corrected * hr_slope + hr_rest
+    hr_ss = np.minimum(hr_max, hr_lin)
+
+    diff_hr = np.pad(np.diff(power_corrected), pad_width=1, mode='constant')
+    rate = np.where((diff_hr <= 0), rate_decay, rate_growth)
+
+    hr_pred = np.ones(power.shape) * hr_rest
+    for curr_idx in enumerate(hr_pred.size):
+        hr_prev = hr_pred[curr_idx] if curr_idx > 0 else hr_rest
```

Comment:
Hmm it appears that this was a bug in my sweatpy code: it always used `hr_rest`. I think you fixed that.